pymc3/distributions/multivariate.py

```diff
@@ -486,23 +486,9 @@ class Dirichlet(Continuous):
 
     def __init__(self, a, transform=transforms.stick_breaking,
                  *args, **kwargs):
-
-        if not isinstance(a, pm.model.TensorVariable):
-            if not isinstance(a, list) and not isinstance(a, np.ndarray):
-                raise TypeError(
-                    'The vector of concentration parameters (a) must be a python list '
-                    'or numpy array.')
-            a = np.array(a)
-            if (a <= 0).any():
-               raise ValueError("All concentration parameters (a) must be > 0.")
-
-        shape = np.atleast_1d(a.shape)[-1]
```

Comment:
Could we have a version that this line still works? Specifying the shape essentially twice looks a bit annoying.

Reply:
How is the shape specified twice?

Reply:
I meant the added `, shape=(...)` in the RV creation, like:
```
w0 = pm.Dirichlet('w0', a=np.ones(2), shape=(2,))
```

Reply:
Ah yeah, could be nice to retain this possibility for `a` with `list` and `np.array` types! IIUC, the shape must be explicitly passed only when `a` is a symbolic variable?

Reply:
> `Dirichlet` was a weird exception.

Like I say in the first comment, `Dirichlet` only *seemed* like an exception, but it never really was.  The way `Dirichlet` was being _used_ was the exception: i.e. only ever with NumPy arrays, and no Theano arguments.

Reply:
> Could we have a version that this line still works? Specifying the shape essentially twice looks a bit annoying.

Unfortunately, no; not without some major rewrites that fix the shape problems completely.  

At best, we could attempt to focus on the number of dimensions instead of the shape, but that will only help with a few things (e.g. we could at least get the `TensorType`s of RVs correct without complete knowledge of the shape).  Even that would be a large change, and any large changes should simply fix the shape issues completely.

Reply:
> IIUC, the shape must be explicitly passed only when `a` is a symbolic variable?

I'm against the idea of filling PyMC3 with more Theano circumvention logic, especially when it doesn't appear to be the norm for the other multivariate distributions.

Reply:
I see your point. Well this is already a pattern in most of the other multivariate distributions (that you need to specify `shape` kwarg). I guess we will need a better solution looking ahead and certain that shouldn't be blocking this.